### DISCUSSION TITLE
Welcome to GirlScript Summer of Code 2021

### DISCUSSION THREAD

**Discussion by vinitshahdeo** [vinitshahdeo/PortScanner#117] [General]
Hello! 👋

**Glad to see you here!**

Last year, [Water Monitoring System](https://github.com/vinitshahdeo/Water-Monitoring-System) project has received great contributions in GSSoC; Grateful to the [contributors](https://github.com/vinitshahdeo/Water-Monitoring-System/issues/15)! 🤗  Here we come again with a new project, [Network Scanner](https://vinitshahdeo.github.io/PortScanner/)! 

Before [GSSoC'21](https://gssoc.girlscript.tech/) begins,

- Please go through the [README.md](https://github.com/vinitshahdeo/PortScanner) and follow the instructions to setup Port Scanner locally. Let us know if you face any problems.

- **Shortly, we'll be opening `beginner-friendly` issues so that you can get the ball rolling once GSSoC kicks off.**

- Here're a few recommended readings:
  - [GSSOC 2020-A journey to explore the “UNEXPLORED”](https://medium.com/girlscript-summer-of-code/gssoc-2020-a-journey-to-explore-the-unexplored-f25a6ade8288) by @Swarnimashukla
  - [GirlScript Summer of Code 2020 Mentor Experience](https://medium.com/girlscript-summer-of-code/girlscript-summer-of-code-2020-mentor-experience-28daec399b1e) by @PragatiVerma18
  - [GO-GREEN featuring GITHUB](https://medium.com/@kashish_121/go-green-featuring-github-f8750fbf0729) by @Kashish121 
  - [git push: You just need a little PUSH](https://medium.com/@ishikadubey2000/look-mom-im-on-github-521bb6c1f51d) by @ishika1727 
  - [If You Are A Techie, Your Home Page Should Be GitHub, Not Instagram](https://www.opensourceforu.com/2020/07/if-you-are-a-techie-your-home-page-should-be-github-not-instagram/) - @vinitshahdeo 

- During these 3 months, our major target is to **migrate the existing frontend codebase to React** and add multiple network scanning features. (💡 **We're open to your ideas, feel free to drop your thoughts here**.)

- Know your mentors:
  - _@vinitshahdeo : Software Engineer at Postman | [LinkedIn](https://www.linkedin.com/in/vinitshahdeo/)_
  - _@Kashish121 : Incoming Software Engineering Intern at Adobe | [LinkedIn](https://www.linkedin.com/in/kashish121/)_
  - _@nandikajain : Incoming STEP Intern at Google | [LinkedIn
](https://www.linkedin.com/in/nandikajain/)_  
  - _@ishika1727 : Incoming SDE Intern at Wells Fargo | [LinkedIn](https://www.linkedin.com/in/ishika1727/)_

- Feel free to ask your doubts here. 

**Exciting times ahead!** ❤️

**Comment by themadhavdihora** [comment]
Hello, this is me madhav dihora, Computer Science and IT Graduate with additional skills of Photography and Videography. Completed my Study from St. Clair College and Charotar University. Currently living in ON, Canada.

I know how important IT, as well as Networking skills with Technical Support in this Era. I consider myself a grounded Technocrat with an entrepreneurial spirit.

My internship experience at DMV IT Solutions and Job SkillShare, has provided me knowledge and experience with Technical Support and Networking skills that work for solving problems.

For more : https://www.madhav.live

**Reply by vinitshahdeo** [reply]
Hello @themadhavdihora,

Glad to e-meet you!
Please have a look into the repo and submit a few feature requests as you already have experience with Networking.

Looking forward to working with you!

**Comment by nitish-awasthi** [comment]
Hi, I'm Nitish Awasthi, currently in my final year of B.Tech in Computer Science and Engineering with Specialization in Cloud Technology and Internet Security. I've started my open source through the Facebook Open Source Immersion Programme and awarded amongst the Top Contributors. 
I've contributed to various open-source projects like RomeFrontEnd, React, Facebook Edu Bot, and many more.
I'm an open-source enthusiast, a passionate coder, and sometimes build small projects for fun.

Welcoming all the fellow mentors and participants.
Special thanks to @vinitshahdeo sir for having me on this project.

You can connect with me here:-
https://www.linkedin.com/in/anitish

**Reply by vinitshahdeo** [reply]
Hey @nitish-awasthi,

Welcome to the clan!
Looking forward to great time learning together. 🚀

**Comment by saaalik** [comment]
Hello, This is Salik, A BSc CS fresher at SSCBS, looking forward to contributing to this project. I am really pumped up
[Connect with me here](https://www.linkedin.com/in/md-salik-75a5021b8/)

**Reply by vinitshahdeo** [reply]
Hey @saaalik,

Glad to see you here!
Wishing you a great open-source journey ahead.

**Comment by TarunaSharma2002** [comment]
Hello coders, I'm Taruna Sharma, pursuing my btech degree in Electronics and Communication stream... I don't have much knowledge of open source, I hope everyone will help each other.

**Reply by vinitshahdeo** [reply]
Hello @TarunaSharma2002,

Glad to see you here.
We'll help you **kickstart your open-source journey**. This [link](https://www.opensourceforu.com/2020/07/if-you-are-a-techie-your-home-page-should-be-github-not-instagram/) might help you. :v:

**Comment by ishika1727** [comment]
Hello!  👋 
This is Ishika Dubey. I am a pre-final year student of B.Tech in Information Technology at VIT, Vellore. I am a web developer and a coder, currently working on React. 

I started my journey of open-source a year ago and it has been amazing collaborating with top contributors. I am elated to see how this project started as a simple Digital Assignment and now, it has become a part of  GSSoC'21. 🎉 

Looking forward to having a great time working with my fellow mentors and participants.

You can connect with me on [LinkedIn](https://www.linkedin.com/in/ishika1727)

A special mention to @vinitshahdeo for having me on this project.

**Reply by vinitshahdeo** [reply]
Hey @ishika1727,

Thanks for all your contributions. 🎉 
Looking forward to many more!

**Comment by ranaaditya** [comment]
Hi everyone, I am Aditya Rana, a prefinal year student at NIT Trichy.

I started open source with GSoC'20 with Processing Foundation where I have written a graphics engine for mobile devices from scratch.
Recently I have developed a keen interest in Networking and Network Security field, so thought why not to mentor few juniors while leaning the stuff.

Thanks to @vinitshahdeo  for this wonderful project.
Hope to learn a lot from other contributors and mentors during this journey.

You can connect with me on [Linkedin](https://www.linkedin.com/in/adityarananitt/)

Thanks !

**Reply by vinitshahdeo** [reply]
Welcome to the team, @ranaaditya!
A lot to learn from you.

**Comment by amanjha21** [comment]
Hello everyone,
I am Aman Jha. I am a 2nd Year student at NIT Jamshedpur.
Looking forward to contributing to this project as a part of GSSOC'21.

**Reply by vinitshahdeo** [reply]
Hello @amanjha21 👋 

Thanks for showing your interest.
Good luck for GSSoC'21.

**Comment by lopeselio** [comment]
Hey this is Elio, looking forward to contribute to PortScanner

**Reply by vinitshahdeo** [reply]
Same here, @lopeselio 👀

**Comment by Abhishek765** [comment]
Hello @vinitshahdeo Glad to connect and happy to contribute to this project I have some react and backend skills, exploring this project....
#loading.....

**Reply by vinitshahdeo** [reply]
Hello @Abhishek765,

Glad to e-meet you!
Looking forward to your awesome contributions.

**Comment by Kashish121** [comment]
Hello everyone! 👋🏽 
I am Kashish, an undergraduate at VIT, Vellore majoring in Information Technology. I am one of the mentors for [Network Scanner](https://github.com/vinitshahdeo/PortScanner) project. So feel free to ping me anytime. 😄 

I started my open source journey as a participant in GSoC'20 contributing in [Water Monitoring System](https://github.com/vinitshahdeo/Water-Monitoring-System) and it's been a super fascinating journey till now! As wisely said by @vinitshahdeo, GitHub is a social media for coding enthusiasts, just like Instagram 😉. I enjoy coming back to GitHub every now and then to take a step deeper into open source.

Find me here : [LinkedIn](https://www.linkedin.com/in/kashish121/). 

Looking forward for some awesome contributions in [Network Scanner](https://github.com/vinitshahdeo/PortScanner). 🌟 

For a complete understanding of the [Network Scanner](https://github.com/vinitshahdeo/PortScanner) project, I strongly recommend reading these two articles:
- [GO-GREEN featuring GITHUB!](https://medium.com/@kashish_121/go-green-featuring-github-f8750fbf0729)
- [git push: You just need a little PUSH!](https://medium.com/@ishikadubey2000/look-mom-im-on-github-521bb6c1f51d)

**Reply by nitish-awasthi** [reply]
Thanks @Kashish121 for sharing these articles. I was searching for something like this.

**Reply by vinitshahdeo** [reply]
Welcome back @Kashish121! 🎉 
Exciting times ahead. 🤩

**Comment by Snehakri022** [comment]
Hi everyone, I am Sneha Kumari, a prefinal year student at JGEC.
Looking forward to contributing to this project as a part of GSSOC'21.

**Reply by vinitshahdeo** [reply]
Thanks @Snehakri022 for showing your interest. Looking forward to your contributions.

**Comment by palrohitg** [comment]
Hi, guys, I am Vikas pal CSE and open-source geek. 
Looking forwards to go deeper into django by scaling portscanner from flask to django :+1: 
And also to provide DevOps infra for the same.

**Reply by vinitshahdeo** [reply]
Yes, FTW @palrohitg 🎉

**Comment by jaiongit** [comment]
Hey everyone! :wave:
I am Jai Juneja, currently a 3rd year B. Tech (CSE) student from a college affiliated to AKTU, Lucknow. This project interests me a lot and I would love to contribute to it while learning along the way. I'm interested to give some valuable contributions to migrate the codebase to React and Django. I know a little bit of Django and completely new to React (as well as to some parts of JavaScript). I would be very happy if the Admin and Mentors could help me in kickstarting my Open Source journey!

**Reply by vinitshahdeo** [reply]
Hey @jaiongit,
We're always there to help you. Wishing you a great journey ahead!

**Reply by jaiongit** [reply]
Thanks!
By the way @vinitshahdeo, I want to know when using Python 3 do we need the exact Python version as specified in the README.md (version 3.8)? :thinking:
As I don't have that version but a more recent version (3.9.1). Also, while trying to run the Flask UI through that same more recent Python version, I'm getting the following error:
```python
File "/path/to/PortScanner/src/mainScanner.py", line 10, in <module>
    from multi.scanner_thread import split_processing
ImportError: bad magic number in 'multi': b'\x03\xf3\r\n'
```
I am assuming this is being caused due to my recent Python version.

PS: Before running the Flask UI, I checked out to `feature/python3.8` branch. Am I doing the right thing?

**Reply by vinitshahdeo** [reply]
@jaiongit Can you please create an issue for the same? 

@Kashish121 will have a look into this.

**Reply by jaiongit** [reply]
Okay. I am on the way for creating the issue.

**Comment by iamsonukushwaha** [comment]
Hello everyone!
I am Sonu Kumar Kushwaha
Looking forward to contribute to this project as a part of GSSOC'21.

**Reply by ranaaditya** [reply]
> Hello everyone!
> I am Sonu Kumar Kushwaha
> Looking forward to contribute to this project as a part of GSSOC'21.

Hi @flyingsonu122 glad to know that you are interested in this awesome project !
You can start contributing by picking up any of these [issues](https://github.com/vinitshahdeo/PortScanner/issues) and feel free to ask doubts !

Thanks,
Network Scanner team.

**Reply by iamsonukushwaha** [reply]
I have already created #128 #129 #130 #131

**Reply by iamsonukushwaha** [reply]
Will create PRs for the same

**Reply by iamsonukushwaha** [reply]
Created a PR #132

**Comment by vinitshahdeo** [comment]
Mentors(@Kashish121, @ishika1727, @nitish-awasthi, @ranaaditya & @themadhavdihora), please ensure that you add `gssoc21` label and the `level` labels to the issues and PRs (before merging).

- `Level0` - Minor Doc Fix like typo. [ 5 Points ]
- `Level1` - Major Doc Fix. [ 10 Points]
- `Level2`- Minor Feature or Bug Solving [ 15 Points ]
- `Level3` - Major Feature or Bug Solving [ 20 points ]

PS: **The scores will be calculated using these labels.**

**Reply by ranaaditya** [reply]
> Mentors(@Kashish121, @ishika1727, @nitish-awasthi, @ranaaditya & @themadhavdihora), please ensure that you add `gssoc21` label and the `level` labels to the issues and PRs (before merging).
> 
>     * `Level0` - Minor Doc Fix like typo. [ 5 Points ]
> 
>     * `Level1` - Major Doc Fix. [ 10 Points]
> 
>     * `Level2`- Minor Feature or Bug Solving [ 15 Points ]
> 
>     * `Level3` - Major Feature or Bug Solving [ 20 points ]
> 
> 
> PS: **The scores will be calculated using these labels.**

Hi @vinitshahdeo , I don't have access to change the labels :(

**Reply by nitish-awasthi** [reply]
> Mentors(@Kashish121, @ishika1727, @nitish-awasthi, @ranaaditya & @themadhavdihora), please ensure that you add `gssoc21` label and the `level` labels to the issues and PRs (before merging).
> 
> * `Level0` - Minor Doc Fix like typo. [ 5 Points ]
> * `Level1` - Major Doc Fix. [ 10 Points]
> * `Level2`- Minor Feature or Bug Solving [ 15 Points ]
> * `Level3` - Major Feature or Bug Solving [ 20 points ]
> 
> PS: **The scores will be calculated using these labels.**

@vinitshahdeo! There's a small update `Level3` issues will be awarded as `25 points`.

**Comment by vinitshahdeo** [comment]
Mentors(@Kashish121, @ishika1727, @nitish-awasthi, @ranaaditya & @themadhavdihora), thanks for your contribution so far. 🙏  Please review the [PRs](https://github.com/vinitshahdeo/PortScanner/pulls) which are opened currently. There are many pending PRs. Close the issues which has been resolved.

> _PS: We'll meet pretty soon for the next steps. Till then, let's clear the long PR history._
